runtime/debug/outer-port-attachment.js

```diff
@@ -22,23 +24,43 @@ export class OuterPortAttachment {
     // Skip speculative and pipes arcs for now.
     if (this._arcIdString.endsWith('-pipes') || this._speculative) return;
 
-    let stack = [];
+    const stack = [];
     if (!isReceiver) {
-      // The slice discards the 'Error' string and the 2 stack frames
-      // corresponding to this function and the API channel function,
-      // which is already being displayed in the log entry.
-      stack = new Error().stack.split('\n    at ').slice(3).map(f => {
-        // Convert 'foo.bar (http://some/url/arcs/shell/sub/Source.js:240:35)'
-        // to {method: 'foo.bar', location: 'shell/sub/Source.js:240'}.
-        // Note that some frames may look like 'Array.forEach (<anonymous>)',
-        // which should still work.
-        const m = f.match(/^(.*) \((.*)\)$/);
-        if (m === null) return {method: f, location: ''};
-        return {
-          method: m[1],
-          location: m[2].replace(/^http.*\/arcs\//, '').replace(/:[0-9]+$/, '')
-        };
-      });
+      // The slice discards the first two stack frames corresponding to this
+      // function and the API channel function, which is already being displayed
+      // in the log entry.
+      mapStackTrace(new Error().stack, mapped => mapped.slice(2).map(f => {
```

Comment:
Maybe worth extracting to its own small file? Can be premature, but it is a standalone functionality that I think will be used more than just here. WDYT?

Reply:
I think until we do need it somewhere else, simplest to just leave it in place.